go/vt/topo/test/testing.go

```diff
@@ -112,3 +122,90 @@ func TopoServerTestSuite(t *testing.T, factory func() *topo.Server) {
 	checkWatchInterrupt(t, ts)
 	ts.Close()
 }
+
+// StartConsul starts a consul subprocess, and waits for it to be ready.
+// Returns the exec.Cmd forked, the config file to remove after the test,
+// and the server address to RPC-connect to.
+func StartConsul(t *testing.T, authToken string) (*exec.Cmd, string, string) {
+	// Create a temporary config file, as ports cannot all be set
+	// via command line. The file name has to end with '.json' so
+	// we're not using TempFile.
+	configDir, err := os.MkdirTemp("", "consul")
+	if err != nil {
+		t.Fatalf("cannot create temp dir: %v", err)
+	}
+	defer os.RemoveAll(configDir)
+
+	configFilename := path.Join(configDir, "consul.json")
+	configFile, err := os.OpenFile(configFilename, os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		t.Fatalf("cannot create tempfile: %v", err)
+	}
+
+	// Create the JSON config, save it.
+	port := testfiles.GoVtTopoConsultopoPort
+	config := map[string]interface{}{
+		"ports": map[string]int{
+			"dns":      port,
+			"http":     port + 1,
+			"serf_lan": port + 2,
+			"serf_wan": port + 3,
+		},
+	}
+
+	if authToken != "" {
+		config["datacenter"] = "vitess"
+		config["acl_datacenter"] = "vitess"
+		config["acl_master_token"] = authToken
+		config["acl_default_policy"] = "deny"
+		config["acl_down_policy"] = "extend-cache"
```

Comment:
these are all deprecated flags, should we consider keeping `startConsul` as is (since it's testing some acl configs) and start consul via a separated minimum cli e.g., `exec.Command("consul", "agent", "-dev")`

Reply:
updated to use its own simplified `startConsul`